ruby_oop/hangman/lib/manage_game_state.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require './display_text'
+
+# module that takes care of saving and loading the saved game
+module ManageGameState
+  include DisplayableText
+
+  FILE_NAME = 'saved_game.yml'
+
+  def load_game_prompt
+    puts DisplayableText::LOAD_GAME_CHOICE
+    @prompt_response = gets.chomp.upcase
+    case @prompt_response
+    when 'Y'
+      puts DisplayableText::LOADED_GAME_CONFIRMATION
+      load_game
+    else
+      puts 'Ok, starting a new game.'
+    end
+  end
+
+  def load_game
+    game_state = YAML.safe_load(File.open(FILE_NAME, 'r'), permitted_classes: [Hangman, RandomWord])
+    @guessed_letters = game_state.guessed_letters
+    @secret_word = game_state.secret_word
+    @hidden_word_lines = game_state.hidden_word_lines
+    @turn_number = game_state.turn_number
+  end
+
+  def save_game_file
+    serialized_object = File.new(FILE_NAME, 'w')
+    serialized_object.puts YAML.dump(self)
+    serialized_object.close
+  end
+
+  def save_the_game_option
+    if @turn_number <= (Hangman::MAX_TURNS - 1) && !win?
+      case @prompt_for_letter
+      when 'save game'
```

Comment:
It's a little odd to have a `case` with just one `when`. Any reason not to use `if @prompt_for_letter == 'save game'`?

Also, it's good practice use a constant for load-bearing strings like `'save game'`.

Reply:
>It's a little odd to have a case with just one when. Any reason not to use if @prompt_for_letter == 'save game'?
I _think_ rubocop complained when I had it as an `if` and proposed a `case`, but maybe I am mixing up rubocop suggestions with personal style choices and it's all a blur in my head. 

I'll update the string to a constant though.